### ISSUE TITLE
Improve documentation website

### ISSUE THREAD

**Issue by Deadpikle** [NetSparkleUpdater/NetSparkle#127]
I wanted to get up a basic docs website so that people could browse properties and such on the various NetSparkleUpdater classes. I used [Natural Docs](https://www.naturaldocs.org/) (quick Google result 😅) to generate some OK docs, and they are up now on the [website here](https://netsparkleupdater.github.io/NetSparkle/doc/index.html). I like that it makes a single page docs app sort of thing. However, I really don't like this setup for a few reasons:

* I have to manually update the docs and push them to the branch that controls the website (`gh-pages`)
* The website is not very pretty.
* The docs aren't very...great....for example, `enum` values are not actually listed out. Maybe there is an option for this. I honestly didn't spend long on this haha

I would like to improve the docs on the website a lot, including integration of the details on this repo's README.md. I don't know what the options are here graphically. The repo also needs to be reconfigured (I think?) so that the docs website is a subfolder of the main repo instead of a separate branch so that CI/CD can generate and update the docs appropriately.

tl;dr: Need to get a better docs website up that can automatically update on CI/CD/repo push.

I'm open to suggestions!
